src/picosvg/svg.py

```diff
@@ -572,27 +638,33 @@ def _transformed_gradient(self, url, transform, shape_bbox):
         # TODO normalize stop elements too
         new_fill.extend(copy.deepcopy(stop) for stop in fill_el)
 
-        fill_el.addnext(new_fill)
+        self._apply_gradient_translation(new_fill)
+
+        self._add_to_defs(defs, new_fill)
         return new_fill
 
     def _simplify(self):
         """
         Removes groups where possible, applies transforms, applies clip paths.
-
-        Groups with 0 < opacity < 1 and multiple children are retained.
         """
         # Reversed: we want leaves first
         to_process = reversed(tuple(c for c in self.breadth_first()))
 
+        defs = etree.Element(f"{{{svgns()}}}defs", nsmap=self.svg_root.nsmap)
+        self.svg_root.insert(0, defs)
+
         for context in to_process:
             if "clipPath" in context.path:
                 _safe_remove(context.element)
                 continue
 
             el = context.element
-            _del_attrs(el, "clip-path", "transform")  # handled separately
+            _del_attrs(el, *_ATTRIB_W_CUSTOM_INHERITANCE)  # handled separately
 
-            skips = _ATTRIB_W_CUSTOM_INHERITANCE | {"opacity"}  # handled separately
+            skips = _ATTRIB_W_CUSTOM_INHERITANCE | _OPACITY_ATTRIB  # handled separately
+            # context.attrib has inherited attrib
+            # if we inherit atop the existing values we'll double-combine, e.g. opacity will *= opacity
```

Comment:
was there a separate bug in addition to #237 where we'd double-combine opacity? if yes, it should figure more prominently in the PR/commit message (and have its own test)